src/libpgagroal/configuration.c

```diff
@@ -2788,13 +2797,50 @@ transfer_configuration(struct configuration* config, struct configuration* reloa
    return 0;
 }
 
+/**
+ * Checks if the configuration of the first server
+ * is the same as the configuration of the second server.
+ * So far it tests for the same connection string, meaning
+ * that the hostname and the port must be the same (i.e.,
+ * pointing to the same endpoint).
+ * It does not resolve the hostname, therefore 'localhost' and '127.0.0.1'
+ * are considered as different hosts.
+ * @return true if the server configurations look the same
+ */
+static bool
+is_same_server(struct server* s1, struct server* s2)
+{
+
+   if(!strncmp(s1->host, s2->host, MISC_LENGTH) && s1->port == s2->port)
+   {
+      return true;
+   }
+   else
+   {
+      return false;
+   }
+}
+
 static void
 copy_server(struct server* dst, struct server* src)
 {
+   atomic_schar state;
+
+   // check the server cloned "seems" the same
+   if (is_same_server(dst,src))
+   {
+      state = atomic_load(&dst->state);
+   }
+   else
+   {
+      state = SERVER_NOTINIT;
```

Comment:
Use atomic_init()

Reply:
shame on me, I did not see the `uncrustify.sh` script!
We should also put a guide or a note on how to contribute to code, e.g., run `uncrustify.sh` before submitting a patch.

Reply:
Last section in `CONTRIBUTING.md`